src/REPLMode/argument_parsers.jl

```diff
@@ -73,7 +73,7 @@ end
 # packages can be identified through: uuid, name, or name+uuid
 # additionally valid for add/develop are: local path, url
 function parse_package_identifier(word::AbstractString; add_or_develop=false)::PackageSpec
-    if add_or_develop && isdir_windows_workaround(expanduser(word))
+    if add_or_develop && (casesensitive_isdir(expanduser(word)) || word in (".", ".."))
```

Comment:
Is the `word in (".", ".."))` stuff needed?

Reply:
Yes. It really wasn't working before, but people couldn't tell because Pkg was mixing paths and urls so freely:
```
julia> Pkg.casesensitive_isdir(".")
false

julia> Pkg.casesensitive_isdir("..")
false
```

The reason for being more strict about repo handling is that we can give better error messages. For example, we used to warn about malformed URLs when a user gave the path to a directory did not exist. Many people were confused by this but I don't think there was ever a formal issue

Reply:
How about `../Foo`, will `casesensitive_isdir` work on that?

Reply:
I'm pretty sure it just does not work on these two special cases

Reply:
Alright, strange. Maybe if we call `abspath` inside `casesensitive_isdir`

Reply:
```
julia> Pkg.casesensitive_isdir("../AppCompiler/")
false

julia> isdir("../AppCompiler/")
true
```

Reply:
I pushed another commit that hopefully fixes this.